h5py/h5t.pyx

```diff
@@ -1994,3 +2004,71 @@ cpdef object py_get_vlen(object dt_in):
     warn("Deprecated; use check_vlen_dtype(dtype) instead",
         H5pyDeprecationWarning)
     return check_vlen_dtype(dt_in)
+
+
+def register_dtype(bytes tag, dtype dt_in):
```

Comment:
I think that the tag should have a default value that is generated from the dtype name.  Thoughts here @takluyver @aragilar